xonsh/aliases.py

```diff
@@ -255,18 +255,19 @@ def source_alias(args, stdin=None):
     env = builtins.__xonsh_env__
     encoding = env.get('XONSH_ENCODING')
     errors = env.get('XONSH_ENCODING_ERRORS')
-    for fname in args:
+    for i, fname in enumerate(args):
         fpath = fname
         if not os.path.isfile(fpath):
             fpath = locate_binary(fname)
-            if fpath is None:
+            if fpath is None and env.get('XONSH_DEBUG'):
```

Comment:
if the path is None and we haven't set `XONSH_DEBUG` this will try to source the file,
is that what we want? the previous behavior seems very different